packages/flutter_tools/test/general.shard/ios/ios_device_start_prebuilt_test.dart

```diff
@@ -0,0 +1,400 @@
+    final FileSystem fileSystem = MemoryFileSystem.test();
+    final FakeProcessManager processManager = FakeProcessManager.list(<FakeCommand>[
+      kDeployCommand,
+      FakeCommand(
+        command: <String>[
+          'ios-deploy',
+          '--id',
+          '123',
+          '--bundle',
+          '/',
+          '--no-wifi',
+          '--justlaunch',
+          // The arguments below are determined by what is passed into
+          // the debugging options argument to startApp.
+          '--args',
+          <String>[
+            '--enable-dart-profiling',
+            '--enable-service-port-fallback',
+            '--disable-service-auth-codes',
+            '--observatory-port=60700',
+            '--start-paused',
+            '--dart-flags="--foo"',
+            '--enable-checked-mode',
+            '--verify-entry-points',
+            '--enable-software-rendering',
+            '--skia-deterministic-rendering',
+            '--trace-skia',
+            '--endless-trace-buffer',
+            '--dump-skp-on-shader-compilation',
+            '--verbose-logging',
+            '--cache-sksl',
+          ].join(' '),
+        ], environment: const <String, String>{
+          'PATH': '/usr/bin:null',
+          'DYLD_LIBRARY_PATH': '/path/to/libraries',
+        }
+      )
+    ]);
+    final IOSDevice device = setUpIOSDevice(
+      sdkVersion: '13.3',
+      processManager: processManager,
+      fileSystem: fileSystem,
+    );
+    final IOSApp iosApp = PrebuiltIOSApp(
+      projectBundleId: 'app',
+      bundleName: 'Runner',
+      bundleDir: fileSystem.currentDirectory,
+    );
+    final Uri uri = Uri(
+      scheme: 'http',
+      host: '127.0.0.1',
+      port: 1234,
+      path: 'observatory',
+    );
+
+    device.setLogReader(iosApp, FakeDeviceLogReader());
+    device.portForwarder = const NoOpDevicePortForwarder();
+
+    when(MDnsObservatoryDiscovery.instance.getObservatoryUri(any, any, usesIpv6: anyNamed('usesIpv6')))
+      .thenAnswer((Invocation invocation) async => uri);
+
+    final LaunchResult launchResult = await device.startApp(iosApp,
+      prebuiltApplication: true,
+      debuggingOptions: DebuggingOptions.enabled(
+        BuildInfo.debug,
+        startPaused: true,
+        disableServiceAuthCodes: true,
+        dartFlags: '--foo',
+        enableSoftwareRendering: true,
+        skiaDeterministicRendering: true,
+        traceSkia: true,
+        traceSystrace: true,
+        endlessTraceBuffer: true,
+        dumpSkpOnShaderCompilation: true,
+        cacheSkSL: true,
+        verboseSystemLogs: true,
+      ),
+      platformArgs: <String, dynamic>{},
+    );
+
+    expect(launchResult.started, true);
+    expect(await device.stopApp(iosApp), false);
+    expect(processManager.hasRemainingExpectations, false);
+  }, overrides: <Type, Generator>{
+    MDnsObservatoryDiscovery: () => MockMDnsObservatoryDiscovery(),
+    Usage: () => MockUsage(),
+  });
+}
+
+IOSDevice setUpIOSDevice({
```

Comment:
I've been using these setup methods instead of a single setUp command. Leads to less state floating around.